File: wifi_info.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void get_wifi_info() {
    #ifdef __linux__
        // Linux: Using nmcli to get Wi-Fi information
        printf("Fetching Wi-Fi networks on Linux...\n");
        system("nmcli -t -f active,ssid,signal,frequency dev wifi");

    #elif _WIN32
        // Windows: Using netsh to get Wi-Fi information
        printf("Fetching Wi-Fi networks on Windows...\n");
        system("netsh wlan show network mode=Bssid");

    #else
        // Unsupported OS
        printf("Unsupported Operating System.\n");
    #endif
}

int main() {
    get_wifi_info();
    return 0;
}

File: wifi_info.py
import subprocess
import platform

def get_wifi_info():
    system_os = platform.system()

    try:
        if system_os == "Linux":
            # Get Wi-Fi info using `nmcli` or `iwconfig`
            result = subprocess.check_output(["nmcli", "-t", "-f", "active,ssid,signal,frequency", "dev", "wifi"], universal_newlines=True)
            print("Available Wi-Fi Networks:\n")
            print(result)

        elif system_os == "Windows":
            # Get Wi-Fi info using `netsh wlan`
            result = subprocess.check_output(["netsh", "wlan", "show", "network", "mode=Bssid"], universal_newlines=True)
            print("Available Wi-Fi Networks:\n")
            print(result)

        else:
            print(f"Unsupported Operating System: {system_os}")

    except subprocess.CalledProcessError as e:
        print("Error while fetching Wi-Fi information.")
        print(e)

if __name__ == "__main__":
    get_wifi_info()
